### PR TITLE
feat(TFD-472): Add List Property into PropertyFactory to support list of string

### PR DESCRIPTION
**Pull request by pbailly**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
Add a two new method on PropertyFactory to support list of String and list of unknown elements.
The list of String will be the most used type of list we are going to use, so adding a specific factory is a nice syntactic sugar.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: daikon/src/main/java/org/talend/daikon/properties/property/PropertyFactory.java [hunk 1]
```diff
@@ -98,6 +99,21 @@ public static StringProperty newString(String name, String initialValue) {
...
 
+    public static TypeLiteral<List<String>> LIST_STRING_ELEMENT = new TypeLiteral<List<String>>() {
+        // This object is here to define the type used in the next property
+    };
+
+    public static Property<List<String>> newStringListProperty(String name) {
+        return new Property<>(LIST_STRING_ELEMENT, name).setValue(new ArrayList<String>());
+    }
+
+    public static <T> Property<List<T>> newListProperty(String name, Class<T> zeEnumType) {
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/properties/property/PropertyFactory.java]: please change the parameter name "zeEnumType" to something else like, zeListType or something else.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/221/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/221/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/225/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/225/)

File: daikon/src/main/java/org/talend/daikon/properties/property/PropertyFactory.java (hunk 1 above)
Comment (pbailly) [inline daikon/src/main/java/org/talend/daikon/properties/property/PropertyFactory.java]: I renamed it "zeListElementType"

**Comment by pbailly** [comment]
Non, it is already correctly supported on the JSON schema.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/250/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/250/)

**Pull request merged**